src/SortFilter/Core.js

```diff
@@ -121,24 +122,35 @@ class Core extends Component {
    * @param {Object} e to the event target of the element
    */
   sortByName = (e) => {
-    e.preventDefault();
-    this.shuffle.filter(e.target.textContent);
+    // this.shuffle.filter(e.target.textContent);
+    const newRelativePathQuery =
+      window.location.pathname + "?" + e.target.textContent.toString();
+    history.pushState(null, "", newRelativePathQuery);
+    this.sortFromUrlSearch();
   }
 
   /**
    * Sort by the specific text
    * @param {string} text
    */
   sortByText = (text) => {
-    this.shuffle.filter(text);
+    // this.shuffle.filter(text);
+    window.location.search = text;
+    this.sortFromUrlSearch();
   }
 
+  sortFromUrlSearch = () => {
+    const searchParams = jsUcfirst(window.location.search.slice(1));
```

Comment:
jsUcFirst is probably not necessary . Its usually not a good practice to use capital letters in the url. See [here](https://webmasters.stackexchange.com/questions/33255/google-and-querystring-parameter-capitalization)

Reply:
It's necessary, cause the filter didn't work with 'city' or 'cat'.
It worked with 'City' or 'Cat'